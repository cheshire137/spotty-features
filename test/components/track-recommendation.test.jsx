import React from 'react'
import renderer from 'react-test-renderer'

import SavedTracksResponse from '../fixtures/spotify/saved-tracks'

import TrackRecommendation from '../../src/components/track-recommendation.jsx'

const item = SavedTracksResponse.items[0]
const track = item.track

function props() {
  return {
    artists: track.artists.map(a => a.name),
    album: track.album.name,
    image: track.album.images[2].url,
    name: track.name,
    url: track.external_urls.spotify,
    albumUrl: track.album.external_urls.spotify
  }
}

describe('TrackRecommendation', () => {
  let component = null

  beforeEach(() => {
    component = <TrackRecommendation {...props()} />
  })

  test('matches snapshot', () => {
    const tree = renderer.create(component).toJSON()
    expect(tree).toMatchSnapshot()
  })
})
