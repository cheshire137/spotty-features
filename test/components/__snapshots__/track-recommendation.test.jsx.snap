exports[`TrackRecommendation matches snapshot 1`] = `
<li>
  <div
    className="columns">
    <div
      className="column track-image-column">
      <a
        className="track-link"
        href="https://open.spotify.com/album/4RPo8gaqAbJksSE8JOIVLF"
        rel="noopener noreferrer"
        target="_blank">
        <img
          className="track-image"
          src="https://i.scdn.co/image/8bf2ccb01c25cf7b314eec154dc0cdf42e3e67df" />
      </a>
    </div>
    <div
      className="column recommended-track-details">
      <a
        className="track-link"
        href="https://open.spotify.com/track/4nDfc6F2uVcc6wdG7kBzWO"
        rel="noopener noreferrer"
        target="_blank">
        <span
          className="track-name">
          Bom Bidi Bom
        </span>
        <span>
          by
        </span>
        <span
          className="track-artists">
          Nick Jonas, Nicki Minaj
        </span>
      </a>
      <a
        className="track-link"
        href="https://open.spotify.com/album/4RPo8gaqAbJksSE8JOIVLF"
        rel="noopener noreferrer"
        target="_blank">
        <span
          className="track-album">
          Fifty Shades Darker (Original Motion Picture Soundtrack)
        </span>
      </a>
    </div>
  </div>
</li>
`;
